rust/lance-core/src/error.rs

```diff
@@ -225,9 +225,19 @@ impl From<std::io::Error> for Error {
 impl From<object_store::Error> for Error {
     #[track_caller]
     fn from(e: object_store::Error) -> Self {
-        Self::IO {
-            source: box_error(e),
-            location: std::panic::Location::caller().to_snafu_location(),
+        match e {
+            object_store::Error::NotFound {
```

Comment:
Hi, I'm afraid this isn't a good fix because the `NotFound` error is so general that there could be many different reasons for encountering it. Converting it to `DatasetNotFound` may cause confusion.

Reply:
Sure - do you have an understanding of what categories the `object_store::Error::NotFound` covers? My impression is that this covers essentially cases where the path passed doesn't point to any actual file, which is more or less the semantics of `DatasetNotFound`. What are the other cases?

Reply:
Hi, it's quite a common error to encounter a `not found` error. For example, users might remove a file from S3, resulting in a `not found` error. In any case, I don't think we should handle this error in this manner.